includes/Admin/UserProfile.php

```diff
@@ -134,13 +134,15 @@ public function add_meta_fields( $user ) {
                     </td>
                 </tr>
 
-                <tr>
-                    <th><?php esc_html_e( 'Store URL', 'dokan-lite' ); ?></th>
-                    <td>
-                        <input type="text" name="dokan_store_url" data-vendor="<?php echo esc_attr( $user->ID ); ?>" class="regular-text" id="seller-url" value="<?php echo esc_attr( $user->data->user_nicename ); ?>"><strong id="url-alart-mgs"></strong>
-                        <p><small><?php echo esc_url( home_url() . '/' . dokan_get_option( 'custom_store_url', 'dokan_general', 'store' ) ); ?>/<strong id="url-alart"><?php echo esc_attr( $user->data->user_nicename ); ?></strong></small></p>
-                    </td>
-                </tr>
+                <?php if ( ! user_can( $user, 'vendor_staff' ) ) : ?>
+                    <tr>
+                        <th><?php esc_html_e( 'Store URL', 'dokan-lite' ); ?></th>
+                        <td>
+                            <input type="text" name="dokan_store_url" data-vendor="<?php echo esc_attr( $user->ID ); ?>" class="regular-text" id="seller-url" value="<?php echo esc_attr( $user->data->user_nicename ); ?>"><strong id="url-alart-mgs"></strong>
+                            <p><small><?php echo esc_url( home_url() . '/' . dokan_get_option( 'custom_store_url', 'dokan_general', 'store' ) ); ?>/<strong id="url-alart"><?php echo esc_attr( $user->data->user_nicename ); ?></strong></small></p>
+                        </td>
+                    </tr>
+                <?php endif; ?>
```

Comment:
Pls remove the `if` condition and generate the url for vendor using `dokan_get_current_user_id()`

```php
$vendor =  new WP_User( dokan_get_current_user_id() );
```
> Now use `$vendor` instead of `$user` to generate the URL.